src/Capability/Registry/ReferenceHandler.php

```diff
@@ -85,10 +85,28 @@ private function prepareArguments(\ReflectionFunctionAbstract $reflection, array
         $finalArgs = [];
 
         foreach ($reflection->getParameters() as $parameter) {
-            // TODO: Handle variadic parameters.
             $paramName = $parameter->getName();
             $paramPosition = $parameter->getPosition();
 
+            // Handle variadic parameters
+            if ($parameter->isVariadic()) {
+                // For variadic parameters, collect all remaining arguments
+                $variadicArgs = [];
+                foreach ($arguments as $key => $value) {
+                    if (is_numeric($key) && $key >= $paramPosition) {
+                        try {
+                            $variadicArgs[] = $this->castArgumentType($value, $parameter);
+                        } catch (InvalidArgumentException $e) {
+                            throw RegistryException::invalidParams($e->getMessage(), $e);
+                        } catch (\Throwable $e) {
+                            throw RegistryException::internalError("Error processing variadic parameter `{$paramName}`: {$e->getMessage()}", $e);
+                        }
```

Comment:
this is basically the same like in line 112-118 - does it make sense to shift the `throw RegistryException` into the `castArgumentType` method instead and skip the `try-catch` in both cases?

or somehow merge this block into lower if-else part as well?